File: String/Dasha_Nightmares.cxx
/*
Codeforces problem 1800F.

Time complexity: O(N²k), where N is the size the number of strings and k in the length of the longest string. 
*/

#include <bits/stdc++.h>
using namespace std;

class Solution {
	public:
	static bool is_nightmare(string a, string b) {
		string t = a + b;
		if(!(t.length()&1)) {
			return false;
		}
		unordered_set<char> occur;
		for(auto x : t) {
			occur.insert(x);
		}
		if(occur.size()!=25) {
			return false;
		}
		unordered_map<char,int> freq;
		for(auto c : t) {
			freq[c]++;
		}
		for(auto x : freq) {
			if(!(x.second&1)) {
				return false;
			}
		}
		return true;
	}
	static int count_pairs(vector<string> words) {
		int count = 0;
		for(auto i(0); i < words.size()-1; ++i) {
			for(auto j(i+1); j < words.size(); ++j) {
				if(is_nightmare(words[i],words[j])) {
					count ++;
				}
			}
		}
		return count;
	}
};

int main(int argc, char **argv) {
	int n;
	cin >> n;
	vector<string> arr;
	string s;

	for(auto i(0); i < n; ++i) {
		cin >> s;
		arr.emplace_back(s);
	}
	cout << "\n" << Solution::count_pairs(arr);
	return 0;
}
		

File: String/Dasha_Nightmares.py
"""
Codeforces problem 1800F.

Time complexity: O(N²k), where N is the size the number of strings and k in the length of the longest string.
"""

from collections import Counter

def is_nightmare(a:str,b:str)->bool:
    t = a+b
    if(len(t)%2==0):
        return False
    occur = set()
    for i in range(len(t)):
        occur.add(t[i])
    if(len(occur)!=25):
        return False
    freq = Counter(t)
    for(key,val) in freq.items():
        if(val%2==0):
            return False
    return True

def count_pairs(words:list)->int:
    count = 0
    for i in range(len(words)-1):
        for j in range(i+1,len(words)):
            if(is_nightmare(words[i],words[j])):
                count+=1
    return count

if __name__=="__main__":
    n = int(input())
    count = 0
    words = []
    for i in range(n):
        s = input()
        words.append(s)
    print(count_pairs(words))
